### DISCUSSION TITLE
Support +inf/-inf for the PgNumeric type

### DISCUSSION THREAD

**Discussion by cmpute** [diesel-rs/diesel#3439] [Ideas]
Positive and negative infinities are introduced in [Postgres 14](https://www.postgresql.org/docs/14/release-14.html). Currently the [`PgNumeric`](https://docs.diesel.rs/master/diesel/pg/data_types/enum.PgNumeric.html) type only supports normal floats and nan.

**Comment by cmpute** [comment]
It's likely going to be a breaking change but it still should be implemented. In case the client is connecting to the server with an older version, a runtime error should be returned for infinities.

**Comment by weiznich** [comment]
Thanks for filling this feature request. I agree that we should somehow support this and I'm open to hear suggestions how to implement this. As a important note: We likely **wont** accept a breaking change for to accept this feature anytime soon as we don't plan to issue another major diesel version anytime soon (== next years).

**Comment by cmpute** [comment]
For a short term fix without break changes, I think it will suffice to create a wrapper like this:
```rust
enum PgNumericWithInf {
    PosInfinity,
    NegInfinity,
    Others(PgNumeric)
}
```
and then implement `FromSql` and `ToSql` for this type as well.

**Reply by weiznich** [reply]
I would accept a PR for this as long as it contains documentation + tests
